clients/bigquery/bigquery.go

```diff
@@ -163,7 +166,63 @@ func (s *Store) putTableViaLegacyAPI(ctx context.Context, tableID TableIdentifie
 }
 
 func (s *Store) putTableViaStorageWriteAPI(ctx context.Context, bqTableID TableIdentifier, tableData *optimization.TableData) error {
-	panic("not implemented")
+	columns := tableData.ReadOnlyInMemoryCols().ValidColumns()
+
+	messageDescriptor, err := columnsToMessageDescriptor(columns)
+	if err != nil {
+		return err
+	}
+	schemaDescriptor, err := adapt.NormalizeDescriptor(*messageDescriptor)
+	if err != nil {
+		return err
+	}
+
+	managedWriterClient, err := managedwriter.NewClient(ctx, bqTableID.ProjectID())
+	if err != nil {
+		return fmt.Errorf("failed to create managedwriter client: %w", err)
+	}
+	defer managedWriterClient.Close()
+
+	managedStream, err := managedWriterClient.NewManagedStream(ctx,
+		managedwriter.WithDestinationTable(
+			managedwriter.TableParentFromParts(bqTableID.ProjectID(), bqTableID.Dataset(), bqTableID.Table()),
+		),
+		managedwriter.WithType(managedwriter.DefaultStream), // TODO: Look into using other stream types
```

Comment:
DefaultStream should be correct https://cloud.google.com/bigquery/docs/write-api#selecting_a_type